### PR TITLE
Add markdown rendering for user notes.

### LINKED ISSUE #28167: Allow Markdown in Task comments
### Description

Implement the support for Markdown in Task notes inside Airflow.

### Use case/motivation

It would be helpful to use markdown syntax in Task notes/comments for the following usecases:
- Formatting headers, lists, and tables to allow more complex note-taking.
- Parsing a URL to reference a ticket in an Issue ticketing system (Jira, Pagerduty, etc.)

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by pierrejeambrun**
closes: #28167.

Allow rendering of markdown for the task and dagrun notes in the grid view.

We use a [react-markdown](https://github.com/remarkjs/react-markdown) which is very well maintained and popular and the [remark-gfm](https://github.com/remarkjs/remark-gfm) plugin to add support for table, strikethrough, checklist and other extra tags.

We add a custom components mapping to be able to render ChakraUI components. This allows us to the markdown part consistent with the rest of the app. (same components). We also have full control here of what is rendered for each element.

A few improvement to consider, that can come as follow up PRs:
- As we render ChakraUI component when rendering the markdown, it is so consistent with the rest of the app that it can be difficult to dissociate what is the note, and what is the actual UI. To address this I added a few style unique to the markdown part so we have some differences (especially in font size) but it is not obvious. (maybe splitting to its own tab would help, see below)
- Long notes are hard to see in the details tab because there is already a lot of information, maybe separating the note in its own tab would make it easier to visualize. (And also dissociate from the rest of the UI)
- During my test, it was working pretty well, and we might sometimes want to put more that 1000 characters in the note. Maybe we could increase this a bit. (A table for instance really fills up the limit quite rapidly, just wondering)
- No support yet for syntax highlighting for code block. This is just a normal 'Code' component and no language specific stuff happen. There are plugins for that, we might want to take a look.
- I activated the 'skipHtml' props, as I don't know yet how safe it is to allow arbitrary html to run in here or if this is working well.

An examples:
image

Another one with a table and a code block:
image

### CODE REVIEW

**Comment by pierrejeambrun** [comment]
Jest is not happy, might need to add the react-markdown module to the babel transformation. I'll take a look

**Review by eladkal — COMMENTED** [review]
~~General comment (it could be applied here but we can also decide it's not relevant at all)~~

~~When we added [instance_name](https://airflow.apache.org/docs/apache-airflow/2.1.2/howto/customize-dag-ui-page-instance-name.html#customizing-dag-ui-header-and-airflow-page-titles) in 2.1 we added two [settings](https://github.com/apache/airflow/blob/2cbe5960476b1f444e940d11177145e5ffadf613/airflow/config_templates/config.yml#L1458-L1467)~~

~~instance_name and `instance_name_has_markup` I wonder if we should change `instance_name_has_markup` to~~ ~~`support_markups` and use generic setting for all places where markdown may be applied `instance_name` or user task comments in notes.~~
~~It just feel odd that sometimes we have a setting and sometimes not?~~

Ignore me. I mixed markdown with markup

**Comment by pierrejeambrun** [comment]
@eladkal I think this can be relevant, I asked myself something similar looking at dag/task documentation where we have different fields `doc`, `doc_json`, `doc_md` for each style. Should we imagine something similar and not mix notes types. I think this is not an issue for now as we only have `text` and `markdown`, but could be if we want to support other markups maybe ?

File: airflow/www/jest.config.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ const config = {
   moduleNameMapper: { // Listing all aliases
     '^src/(.*)$': '<rootDir>/static/js/$1',
   },
+  transformIgnorePatterns: [], // transform node_modules
```
Comment (bbovenzi) [inline airflow/www/jest.config.js]: @pierrejeambrun were you going to update this array to only ignore the specific files from react-markdown?

Comment (pierrejeambrun) [reply airflow/www/jest.config.js]: Just updated the PR !

**Review by bbovenzi — APPROVED** [review]

**Pull request merged**